build/sonic_build_script.sh

```diff
@@ -0,0 +1,346 @@
+do
+    usep=$(echo "$output" | awk '{ print $1}' | cut -d'%' -f1 )
+    if [ $usep -ge $ALERT ]; then
+        echo "Not enough space. Please check build server to free space."
+        exit 1
+    fi
+done
+}
+
+cleanup_server()
+{
+    # Remove all stopped containers
+    docker system prune -a --volumes -f
+
+    # check for disk space and cleanup
+    check_free_space . 65
+}
+
+clone_ws()
+{
+    # Clone the Sonic source code
+    SONIC_SOURCE_DIR=ABT-$BRANCH-$REL_BUILD_TSTAMP
+    sudo rm -rf $SONIC_SOURCE_DIR
+    mkdir $SONIC_SOURCE_DIR
+    cd $SONIC_SOURCE_DIR
+    if [ -v NO_DEL_WS ]; then
+        touch no_del_ws
+    fi
+    git clone -b $BRANCH $GIT_HUB_URL
+    check_error $? "git-clone"
+    cd sonic-buildimage
+    if [ -v BRANCH_COMMIT ]; then
+        git checkout $BRANCH_COMMIT
+    fi
+    make init
+    check_error $? "make_init"
+    git log -1 > commit_log.txt
+}
+
+patch_ws()
+{
+    if [ -v PATCH_SCRIPT_URL ]; then
+        wget --timeout=2 -c $PATCH_SCRIPT_URL
+        URL=${PATCH_SCRIPT_URL%marvell_sonic_patch_script.sh}
+        if [ "$BRANCH" == "master" ]; then
+            PATCH_BRANCH="master-bookworm"
+        else
+            PATCH_BRANCH=${BRANCH}
+        fi
+        bash marvell_sonic_patch_script.sh --branch ${PATCH_BRANCH} --platform ${BUILD_PLATFORM} --arch ${BUILD_PLATFORM_ARCH} --url ${URL}
+        check_error $? "patch_script"
+    fi
+}
+
+build_ws()
+{
+    local startTime=$SECONDS
+
+    # Set the build options
+    mkdir -p $CACHE_DIR/$BRANCH/$BUILD_PLATFORM_ARCH
+    BUILD_OPTIONS=""
+    if [ "$NO_CACHE" == "N" ]; then
+        BUILD_OPTIONS="DEFAULT_CONTAINER_REGISTRY=${MIRROR} SONIC_VERSION_CONTROL_COMPONENTS=${VERSION_CONTROL_COMPONENTS} SONIC_DPKG_CACHE_METHOD=rwcache SONIC_DPKG_CACHE_SOURCE=$CACHE_DIR/$BRANCH/$BUILD_PLATFORM_ARCH/"
+    fi
+    if [ "$BUILD_RPC" == "Y" ]; then
+        if [ "$BUILD_PLATFORM_ARCH" != "armhf" ]; then
+            BUILD_OPTIONS="${BUILD_OPTIONS} ENABLE_SYNCD_RPC=y"
+        fi
+    fi
+    if [[ ! -z ${OTHER_BUILD_OPTIONS} ]]; then
+        BUILD_OPTIONS="${BUILD_OPTIONS} ${OTHER_BUILD_OPTIONS}"
+    fi
+    if [[ ! -z ${ADMIN_PASSWORD} ]]; then
+        BUILD_OPTIONS="${BUILD_OPTIONS} DEFAULT_PASSWORD=${ADMIN_PASSWORD}"
+    fi
+
+    # Build Sonic
+    echo $BUILD_OPTIONS > build_args.txt
+    if [ "$BUILD_PLATFORM_ARCH" == "amd64" ]; then
+        ENABLE_DOCKER_BASE_PULL=y make configure PLATFORM=${BUILD_PLATFORM} $BUILD_OPTIONS
+        check_error $? "configure"
+        make $BUILD_OPTIONS target/sonic-${BUILD_PLATFORM}.bin
+        check_error $? "sonic-${BUILD_PLATFORM}.bin"
+    else
+        ENABLE_DOCKER_BASE_PULL=y make configure PLATFORM=${BUILD_PLATFORM} PLATFORM_ARCH=${BUILD_PLATFORM_ARCH} $BUILD_OPTIONS
+        check_error $? "configure"
+        make $BUILD_OPTIONS target/sonic-${BUILD_PLATFORM}-${BUILD_PLATFORM_ARCH}.bin
+        check_error $? "sonic-${BUILD_PLATFORM}-${BUILD_PLATFORM_ARCH}.bin"
```

Comment:
Can we save the build logs which may be helpful to know which stage the build has failed ?

Reply:
Yes. Will do it.

Reply:
Seeing some issue with this.. Will try to do in next iteration.